### PR TITLE
core: add cz-customizable to establish a commit message convention

### PR DESCRIPTION
**Pull request by paulirish**
image

image

To use:
`yarn global add commitizen`
`git add ....`
`git cz`

Only necessary for first commit in a branch, to set the PR commit name. (Since we squash) followup commits can be whatever.

----------

Later we can look at getting https://github.com/marionebl/commitlint running in a github commit status bot to check our PR titles.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -109,5 +111,13 @@
...
     }
   ],
-  "homepage": "https://github.com/GoogleChrome/lighthouse#readme"
+  "homepage": "https://github.com/GoogleChrome/lighthouse#readme",
+  "config": {
+    "commitizen": {
+      "path": "./node_modules/cz-customizable"
+    },
+    "cz-customizable": {
+      "config": "./.cz-config.js"
```
Comment (brendankenny) [inline package.json]: think you forgot this file :)

Comment (paulirish) [reply package.json]: indeed.

**Comment by paulirish** [comment]
@brendankenny @patrickhulce wdyt

File: .cz-config.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+// ❯ new-audit:     A new audit
+// core:     (Non-new) audits, gatherers, driver, LHR JSON, etc
+// tests:     Tests, smokehouse, etc
+// docs:    Documentation
+// deps:    Dependency changes
+// report:  Report, UI, renderers
+// cli:  CLI stuff
+// extension:  Chrome extension stuff
+// misc:   Something else
```
Comment (brendankenny) [inline .cz-config.js]: new line

File: .cz-config.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+'use strict';
```
Comment (brendankenny) [inline .cz-config.js]: license header since it's js

**Comment by paulirish** [comment]
@googlebot u high, bro?

File: .cz-config.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+module.exports = {
+  allowBreakingChanges: ['core'],
+  allowCustomScopes: true,
+  scopes: [],
+  types: [
+    {value: 'new-audit',  name: 'new-audit: A new audit'},
+    {value: 'core',       name: 'core:      Driver, gather, (non-new) audits, LHR JSON, etc'},
+    {value: 'tests',      name: 'tests:     Tests, smokehouse, etc'},
+    {value: 'docs',       name: 'docs:      Documentation'},
+    {value: 'deps',       name: 'deps:      Dependency bumps only'},
```
Comment (patrickhulce) [inline .cz-config.js]: I was thinking about this one, there are frequently dependency bumps included with other changes, do we want to enforce a separate PR for it?
